.github/workflows/package.yml

```diff
@@ -57,18 +63,15 @@ jobs:
 
     - name: Sign
       env:
-        CERTIFICATE_BASE64: ${{ secrets.CODE_SIGNING_CERT_BASE64 }}
-        CERTIFICATE_PASSWORD: ${{ secrets.CODE_SIGNING_PASSWORD }}
-      run: |
-        # create pfx from base64
-        $pfxPath = Join-Path -Path $env:RUNNER_TEMP -ChildPath "cert.pfx";
-        $certBytes = [System.Convert]::FromBase64String($env:CERTIFICATE_BASE64)
-        $certCollection = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2Collection
-        $certCollection.Import($certBytes, $env:CERTIFICATE_PASSWORD, [System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::Exportable)
-        $protectedCertificateBytes = $certCollection.Export([System.Security.Cryptography.X509Certificates.X509ContentType]::Pkcs12, $env:CERTIFICATE_PASSWORD)
-        [System.IO.File]::WriteAllBytes($pfxPath, $protectedCertificateBytes)
+        AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+        AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+        AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+        CODE_SIGNING_CERT_NAME: ${{ vars.CODE_SIGNING_CERT_NAME }}
+        CODE_SIGNING_VAULT_URI: ${{ vars.CODE_SIGNING_VAULT_URI }}
 
-        dotnet nuget sign src/Biosero.Orchestrator.ScriptingTools/bin/${{ env.BUILD_CONFIGURATION }}/Biosero.Orchestrator.ScriptingTools.${{ steps.gitversion.outputs.fullSemVer }}.nupkg --certificate-path "$pfxPath" --certificate-password "$env:CERTIFICATE_PASSWORD" --timestamper http://timestamp.digicert.com
+      run: |
+        $file = "src/Biosero.Orchestrator.ScriptingTools/bin/${{ env.BUILD_CONFIGURATION }}/Biosero.Orchestrator.ScriptingTools.${{ steps.gitversion.outputs.fullSemVer }}.nupkg"
+        NuGetKeyVaultSignTool sign --file-digest sha256 --timestamp-rfc3161 http://timestamp.digicert.com --timestamp-digest sha256 --azure-key-vault-url ${{ vars.CODE_SIGNING_VAULT_URI }} --azure-key-vault-client-id  ${{ secrets.AZURE_CLIENT_ID }} --azure-key-vault-tenant-id ${{ secrets.AZURE_TENANT_ID }} --azure-key-vault-client-secret ${{ secrets.AZURE_CLIENT_SECRET }} --azure-key-vault-certificate ${{ vars.CODE_SIGNING_CERT_NAME }} "$file"
```

Comment:
Just a heads up that when this tool fails it unfortunately fails silently and its usually due to the file not being found in the location provided. Also we need to introduce the dll signing to this workflow as well after the build step.